### PR TITLE
convert moment with timezone to UTC instead of raising an exception

### LINKED ISSUE #29576: DateTimeSensorAsync breaks if target_time is timezone-aware
### Apache Airflow version

2.5.1

### What happened

`DateTimeSensorAsync` fails with the following error if `target_time` is aware:
```
[2022-06-29, 05:09:11 CDT] {taskinstance.py:1889} ERROR - Task failed with exception
Traceback (most recent call last):a
  File "/opt/conda/envs/production/lib/python3.9/site-packages/airflow/sensors/time_sensor.py", line 60, in execute
    trigger=DateTimeTrigger(moment=self.target_datetime),
  File "/opt/conda/envs/production/lib/python3.9/site-packages/airflow/triggers/temporal.py", line 42, in __init__
    raise ValueError(f"The passed datetime must be using Pendulum's UTC, not {moment.tzinfo!r}")
ValueError: The passed datetime must be using Pendulum's UTC, not Timezone('America/Chicago')
```

### What you think should happen instead

Given the fact that `DateTimeSensor` correctly handles timezones, this seems like a bug.  `DateTimeSensorAsync` should be a drop-in replacement for `DateTimeSensor`, and therefore should have the same timezone behavior.

### How to reproduce

```
#!/usr/bin/env python3
import datetime

from airflow.decorators import dag
from airflow.sensors.date_time import DateTimeSensor, DateTimeSensorAsync
import pendulum


@dag(
    start_date=datetime.datetime(2022, 6, 29),
    schedule='@daily',
)
def datetime_sensor_dag():
    naive_time1 = datetime.datetime(2023, 2, 16,  0,  1)
    aware_time1 = datetime.datetime(2023, 2, 16,  0,  1).replace(tzinfo=pendulum.local_timezone())
    naive_time2 = pendulum.datetime(2023, 2, 16, 23, 59)
    aware_time2 = pendulum.datetime(2023, 2, 16, 23, 59).replace(tzinfo=pendulum.local_timezone())

    DateTimeSensor(task_id='naive_time1', target_time=naive_time1, mode='reschedule')
    DateTimeSensor(task_id='naive_time2', target_time=naive_time2, mode='reschedule')
    DateTimeSensor(task_id='aware_time1', target_time=aware_time1, mode='reschedule')
    DateTimeSensor(task_id='aware_time2', target_time=aware_time2, mode='reschedule')

    DateTimeSensorAsync(task_id='async_naive_time1', target_time=naive_time1)
    DateTimeSensorAsync(task_id='async_naive_time2', target_time=naive_time2)
    DateTimeSensorAsync(task_id='async_aware_time1', target_time=aware_time1)  # fails
    DateTimeSensorAsync(task_id='async_aware_time2', target_time=aware_time2)  # fails

datetime_sensor_dag()
```

This can also happen if the `target_time` is naive and `core.default_timezone = system`.

### Operating System

CentOS Stream 8

### Versions of Apache Airflow Providers

N/A

### Deployment

Other

### Deployment details

Standalone

### Anything else

This appears to be nearly identical to #24736.  Probably worth checking other time-related sensors as well.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #29576 

---
Currently, when we provide a datetime with timezone different from UTC to `DateTimeTrigger`, it raises an exception. Instead, I convert the datetime to UTC.

### CODE REVIEW

File: airflow/triggers/temporal.py [hunk 1]
```diff
@@ -41,10 +41,8 @@ def __init__(self, moment: datetime.datetime):
         # Make sure it's in UTC
         elif moment.tzinfo is None:
             raise ValueError("You cannot pass naive datetimes")
-        elif not hasattr(moment.tzinfo, "offset") or moment.tzinfo.offset != 0:  # type: ignore
-            raise ValueError(f"The passed datetime must be using Pendulum's UTC, not {moment.tzinfo!r}")
         else:
-            self.moment = moment
+            self.moment = timezone.convert_to_utc(moment)
```
Comment (Taragolis) [inline airflow/triggers/temporal.py:45]: "Tests Police" here 🤣 , we need also test this one.

A guess we do not have unittests for this constructor except type checking.
The same valid for `TimeDeltaTrigger` which based on `DateTimeTrigger`

Comment (hussein-awala) [reply airflow/triggers/temporal.py:45]: I just updated the test `test_datetime_trigger_timing` to check if the logic remains valid with different timezones

Comment (Taragolis) [reply airflow/triggers/temporal.py:45]: I guess we could check just in class constructor but this seems like even better

**Review by Taragolis — APPROVED** [review]
Wait till CI green

**Review by dimberman — APPROVED** [review]

File: airflow/triggers/temporal.py (hunk 1 above)
Comment (dimberman) [reply airflow/triggers/temporal.py:45]: @Taragolis looks like there are tests added. Do you think there should be more testing?

**Pull request merged**